probe-rs/src/debug/mod.rs

```diff
@@ -2444,18 +2484,19 @@ impl<'debuginfo> UnitInfo<'debuginfo> {
         stack_frame_registers: &Registers,
         cache: &mut VariableCache,
     ) -> Result<Variable, DebugError> {
-        child_variable.type_name = match node.entry().attr(gimli::DW_AT_name) {
-            Ok(optional_name_attr) => match optional_name_attr {
-                Some(name_attr) => extract_name(self.debug_info, name_attr.value()),
-                None => "<unnamed type>".to_owned(),
-            },
+        let type_name = match node.entry().attr(gimli::DW_AT_name) {
+            Ok(optional_name_attr) => {
+                optional_name_attr.map(|name_attr| extract_name(self.debug_info, name_attr.value()))
+            }
             Err(error) => {
                 let message = format!("Error: evaluating type name: {:?} ", error);
                 child_variable.set_value(VariableValue::Error(message.clone()));
-                message
+                Some(message)
             }
         };
+
         if child_variable.is_valid() {
+            /*
```

Comment:
I can't find where you've implemented this logic. In fact, in my tests, the other structured variables are no longer 'lazy' loaded. Can you please clarify the new logic?

Reply:
I don't understand how the handling of "lazy" variables should work, since all variables seem to be resolved right now. For me, this is no different to the current behaviour on master.

Do you have an example where this PR works differently than master?

Reply:
@Tiwalun, I'm still getting differences between the master and this PR for this example. Specifically it fails to resolve the third level `None` value ...

On master ...
<img width="437" alt="master" src="https://user-images.githubusercontent.com/5413046/158877105-a47c3179-6834-4016-8774-3404f49429c9.png">

With this PR ...
<img width="692" alt="this_PR" src="https://user-images.githubusercontent.com/5413046/158877165-df2cb6f7-9e29-4553-bd44-3e6fad778497.png">